src/github.com/getlantern/lantern-mobile/app/src/main/java/org/getlantern/lantern/fragment/FeedFragment.java

```diff
@@ -62,28 +55,48 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container,
         View view = inflater.inflate(R.layout.feed_fragment, container, false);
 
         mList = (ListView)view.findViewById(R.id.feed);
-        adapter = new FeedAdapter(getActivity(), mFeedItems);
         mList.setAdapter(adapter);
         return view;
     }
 
+    protected class LoadFeed extends AsyncTask<String, Void, List<FeedItem>> {
+
+        @Override
+        protected List<FeedItem> doInBackground(String... params) {
+
+            String name = params[0];
+
+            final List<FeedItem> items = new ArrayList<FeedItem>();
+
+            Lantern.FeedByName(name, new Lantern.FeedRetriever.Stub() {
+                public void AddFeed(String title, String desc, 
+                        String image, String url) {
+                    items.add(new FeedItem(title, desc, image, url));
+                }
+            });
+
+            return items;
+        }
+
+        @Override
+        protected void onPostExecute(List<FeedItem> items) {
+            super.onPostExecute(items);
+            mFeedItems.clear();
+            mFeedItems.addAll(items);
+
+            Log.d(TAG, String.format("Feed %s has %d items", feedName, 
+                        items.size()));
+            if (adapter != null) {
+                adapter.notifyDataSetChanged();
+            }
+        }
+
+    }
+
     @Override
     public void onViewCreated(View view, Bundle savedInstanceState) {
         super.onViewCreated(view, savedInstanceState);
-        Log.d(TAG, "Created view for " + this.feedName);
-        new Thread() {
-            public void run() {
-                final List<FeedItem> items = new ArrayList<FeedItem>();
-
-                Lantern.FeedByName(feedName, new Lantern.FeedRetriever.Stub() {
-                    public void AddFeed(String title, String desc, 
-                            String image, String url) {
-                        items.add(new FeedItem(title, desc, image, url));
-                    }
-                });
-
-                NotifyDataSetChanged(items);
-            }
-        }.start();
+        Log.d(TAG, "onViewCreated for " + this.feedName);
+        new LoadFeed().execute(this.feedName);
```

Comment:
I'm worried about the feedName being potentially null or "" here @atavism. Is that possible?

Also, a unit test would be pretty amazing here.